dio/test/options_test.dart

```diff
@@ -5,8 +5,32 @@ import 'package:dio/dio.dart';
 import 'package:test/test.dart';
 
 import 'mock/adapters.dart';
+import 'utils.dart';
 
 void main() {
+  setUp(startServer);
+  tearDown(stopServer);
+
+  test('headers maintained after redirects', () async {
+    final dio = Dio(
+      BaseOptions(
+        baseUrl: serverUrl.toString(),
+        headers: {'x-test-base': 'test-base'},
+      ),
+    );
+    final r = await dio.get(
```

Comment:
shouldn't variable `r` be named in full to something line `final response = await ....` like response